src/v/cluster/partition_probe.cc

```diff
@@ -290,6 +290,13 @@ void replicated_partition_probe::setup_public_metrics(const model::ntp& ntp) {
            topic_label(ntp.tp.topic()),
            partition_label(ntp.tp.partition())})
           .aggregate({sm::shard_label, partition_label}),
+        sm::make_histogram(
+          "read_distribution",
+          [this] { return _read_distribution.read_dist_histogram_logform(); },
+          sm::description("Read path time distribution histogram"),
+          labels)
+          .aggregate({sm::shard_label, partition_label}),
```

Comment:
I think we need to also aggregate the topic here. Otherwise public metrics will massively explode for many-topic clusters.

Side question, is there a reason we put this in public metrics?

Reply:
I think that per-topic read distributions are probably a better source of information than one aggregated histogram, but I do agree that the histogram should be moved to internal metrics. Will fix.